modules/log/groutinelabel.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package log
+
+import "unsafe"
+
+//go:linkname runtime_getProfLabel runtime/pprof.runtime_getProfLabel
+func runtime_getProfLabel() unsafe.Pointer // nolint
+
+type labelMap map[string]string
+
+func getGoroutineLabels() map[string]string {
+	l := (*labelMap)(runtime_getProfLabel())
+	if l == nil {
+		return map[string]string{}
```

Comment:
Some simple unit test could be used to make sure getGoroutineLabels works as expection.

And if `runtime_getProfLabel` is nil, I think `return nil` is better than `return map[string]string{}`  because the LabelMap is indeed nil.

Reply:
I actually like the approach to supply an empty map more because you always have at most one check, namely if the key exists in the map, and not the additional check if the map exists.

Reply:
Nope, you see, the code below is `labels := getGoroutineLabels();	if labels != nil {`

When you are writing Go code, you should always check if a map is nil before using it by keys.

Reply:
Yep, you are correct, there is an additional nil check.
However, I have no idea why.
Because as far as I can see, in this case, it simply cannot return nil.
We could add a comment that the returned map will never be nil, or we can do your approach. I'm fine with both.

Reply:
> We could add a comment that the returned map will never be nil

How about this one .... "And since `l` is `*map`, the `return *l` below might be nil too", I think there is no guarantee that the returned map is non-nil.

And one more thing, checking `m != nil` is cheaper than a new allocation for `map[string]string{}`.